### PR TITLE
discovery: Price checks when caching orchs

### PR DESCRIPTION
**Pull request by yondonfu**
**What does this pull request do? Explain your changes. (required)**

Adds price checks when caching orchestrator responses.

**Specific updates (required)**

See commit history.

**How did you test each of these updates (required)**

Tested manually and added unit tests.

**Does this pull request close any open issues?**

N/A

**Checklist:**

- [x] Read the [contribution guide](./doc/contributing.md)
- [x] `make` runs successfully
- [x] All tests in `./test.sh` pass
- [x] README and other documentation updated
- [x] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

**Comment by yondonfu** [comment]
> It's really hard to tell which orch had the problem, add to log message?

Fixed in https://github.com/livepeer/go-livepeer/pull/1814/commits/4313e327322af00adc71149a6f22246fc9d4869c.

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #### Broadcaster
 
 - \#1782 Fix SegsInFlight data-loss on refreshing O sessions (@darkdragon)
+- \#1814 Add price checks when caching orchestrator responses during discovery
```
Comment (kyriediculous) [inline CHANGELOG_PENDING.md]: Think it is missing an author here , looks good otherwise !

Comment (yondonfu) [reply CHANGELOG_PENDING.md]: Heh realized right after I reviewed your PR 😆 

Fixed in c4cefe3

**Comment by kyriediculous** [comment]
LGTM after squashing 👍

**Review by kyriediculous — APPROVED** [review]

**Pull request merged**